pkg/e2e/operators/cloudingress/public_private.go

```diff
@@ -98,4 +109,15 @@ func updateApplicationIngress(h *helper.H, lbscheme string) {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+func getingressController(h *helper.H, name string) (operatorv1.IngressController, *unstructured.Unstructured) {
+	var ingressController operatorv1.IngressController
+	ingresscontroller, err := h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Get(context.TODO(), name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(ingresscontroller.Object, &ingressController)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
optional: Instead of checking the errors inside the function, we can return them and check in the main method.
This would also reduce number of `Expect(err).NotTo(HaveOccurred())` used in general